thrust/detail/config/compiler.h

```diff
@@ -71,10 +71,10 @@
 #define THRUST_DEVICE_COMPILER THRUST_DEVICE_COMPILER_GCC
 #elif THRUST_HOST_COMPILER == THRUST_HOST_COMPILER_CLANG
 // CUDA-capable clang should behave similar to NVCC.
-#if defined(__CUDA__)
-#define THRUST_DEVICE_COMPILER THRUST_DEVICE_COMPILER_NVCC
-#else
+#if defined(__CUDA__) && defined(__clang__)
```

Comment:
Furthermore, if THRUST_HOST_COMPILER == THRUST_HOST_COMPILER_CLANG, then it will trigger static asserts in number of places that expect compiler for CUDA code to be NVCC.
For instance there's a static assert in thrust/system/cuda/detail/for_each.inl:109